Mathlib/Order/OrderIsoNat.lean

```diff
@@ -49,54 +49,49 @@ def natGT (f : ℕ → α) (H : ∀ n : ℕ, r (f (n + 1)) (f n)) : ((· > ·) :
 theorem coe_natGT {f : ℕ → α} {H : ∀ n : ℕ, r (f (n + 1)) (f n)} : ⇑(natGT f H) = f :=
   rfl
 
-theorem exists_not_acc_lt_of_not_acc {a : α} {r} (h : ¬Acc r a) : ∃ b, ¬Acc r b ∧ r b a := by
-  contrapose! h
-  refine ⟨_, fun b hr => ?_⟩
-  by_contra hb
-  exact h b hb hr
+@[deprecated (since := "2025-08-08")]
+alias exists_not_acc_lt_of_not_acc := exists_not_acc_lt_of_not_acc
 
 /-- A value is accessible iff it isn't contained in any infinite decreasing sequence. -/
-theorem acc_iff_no_decreasing_seq {x} :
-    Acc r x ↔ IsEmpty { f : ((· > ·) : ℕ → ℕ → Prop) ↪r r // x ∈ Set.range f } := by
-  constructor
-  · refine fun h => h.recOn fun x _ IH => ?_
-    constructor
-    rintro ⟨f, k, hf⟩
-    exact IsEmpty.elim' (IH (f (k + 1)) (hf ▸ f.map_rel_iff.2 (Nat.lt_succ_self _))) ⟨f, _, rfl⟩
-  · have : ∀ x : { a // ¬Acc r a }, ∃ y : { a // ¬Acc r a }, r y.1 x.1 := by
-      rintro ⟨x, hx⟩
-      cases exists_not_acc_lt_of_not_acc hx with
-      | intro w h => exact ⟨⟨w, h.1⟩, h.2⟩
-    choose f h using this
-    refine fun E =>
-      by_contradiction fun hx => E.elim' ⟨natGT (fun n => (f^[n] ⟨x, hx⟩).1) fun n => ?_, 0, rfl⟩
-    simp only [Function.iterate_succ']
-    apply h
-
-theorem not_acc_of_decreasing_seq (f : ((· > ·) : ℕ → ℕ → Prop) ↪r r) (k : ℕ) : ¬Acc r (f k) := by
-  rw [acc_iff_no_decreasing_seq, not_isEmpty_iff]
+theorem acc_iff_isEmpty_subtype_mem_range {x} :
+    Acc r x ↔ IsEmpty { f : ((· > ·) : ℕ → ℕ → Prop) ↪r r // x ∈ Set.range f } where
+  mp acc := .mk fun ⟨f, k, hk⟩ ↦ not_acc_iff_exists_descending_chain.mpr
+    ⟨(f <| k + ·), hk, fun _n ↦ f.map_rel_iff.2 (Nat.lt_succ_self _)⟩ acc
+  mpr h := of_not_not fun nacc ↦
+    have ⟨f, hf⟩ := not_acc_iff_exists_descending_chain.mp nacc
+    h.elim ⟨natGT f hf.2, 0, hf.1⟩
+
+theorem not_acc (f : ((· > ·) : ℕ → ℕ → Prop) ↪r r) (k : ℕ) : ¬Acc r (f k) := by
```

Comment:
I think this, and `not_wellFounded`, could have more descriptive names, how about `not_acc_of_descending_chain`?

Reply:
In fact I think it's cleaner to use the idea https://github.com/leanprover-community/mathlib4/pull/28120/files#r2264636988 here more uniformly, which would suggest that name too.

Reply:
I just feel a bit repetitive to have both RelEmbedding and descending_chain in the name. Do you think `not_acc_of_natGT` is better? `natGT` is basically the only info missing from `RelEmbedding`, and there's an actual decl named [RelEmbedding.natGT](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/OrderIsoNat.html#RelEmbedding.natGT).

Reply:
Ah, I forgot that these are in the RelEmbedding namespace; in that case these should be fine

Reply:
I think it's fine as-is